app/kumactl/pkg/entities/entities.go

```diff
@@ -0,0 +1,32 @@
+package entities
+
+import (
+	core_system "github.com/kumahq/kuma/pkg/core/resources/apis/system"
+	"github.com/kumahq/kuma/pkg/core/resources/model"
+)
+
+type Definition struct {
+	Singular     string
+	Plural       string
+	ResourceType model.ResourceType
+	ReadOnly     bool
+}
```

Comment:
This is the same as `ResourceWsDefinition`. Can we just generate this type information in one package and consume it from both kumactl and the API server?

Reply:
So I'm trying to get to the bottom of kumactl/kds/ws and then see what can be done to get everything together (maybe without any codegen and only protobuf options). I feel like it will be easier to build the right thing once we have all the different bits implemented already.
WDYT?

Reply:
Yeh, I agree that this approach seems pretty clean. I think I would have preferred it in a PR by itself, but that's not a big deal really.